src/games/seabattle/ai.ts

```diff
@@ -0,0 +1,191 @@
+import { IAIConfig } from '../index';
+import { ICell, IShip, ISalvo, generateRandomShips } from './game';
+
+interface IAISalvo extends ISalvo {
+  up: boolean;
+  down: boolean;
+  left: boolean;
+  right: boolean;
+}
+
+interface IPlayState {
+  G: any;
+  ctx: any;
+}
+
+class SeabattleBot {
+  async play(state: IPlayState, playerID: string) {
+    if (state.ctx.phase === 'setup') {
+      const shipPositions = generateRandomShips(1);
+      return this.makeSetShipsMove(shipPositions);
+    } else {
+      const cell = this.generateMove(state);
+      return this.makeSalvoMove(cell);
+    }
+  }
+
+  makeSetShipsMove(ships: IShip[]) {
+    return { action: { type: 'MAKE_MOVE', payload: { type: 'setShips', args: [ships], playerID: '1' } } };
+  }
+
+  makeSalvoMove(cell: ICell) {
+    return { action: { type: 'MAKE_MOVE', payload: { type: 'salvo', args: [cell.x, cell.y], playerID: '1' } } };
+  }
+
+  generateMove(state: IPlayState) {
+    const salvos = state.G.salvos.filter((salvo: ISalvo) => (
+      salvo.player === 1 &&
```

Comment:
this salvo.player === 1 should probably not be hard coded... you have the playerID coming from play() and you could pass into the generateMove

Reply:
otherwise the Simulate wont work

Reply:
I actually noticed this the other day, fixed it in TicTacToe's AI, and intended to fix in Seabattle but forgot.  https://github.com/freeboardgame/FreeBoardGame.org/pull/175 :-p